pkg/controller/servicebinding/secret.go

```diff
@@ -46,37 +83,38 @@ func (s *secret) createOrUpdate(payload map[string][]byte, ownerReference metav1
 		return nil, err
 	}
 
-	resourceClient := s.buildResourceClient()
-
 	logger.Debug("Attempt to create secret...")
 	existingSecret, err := s.get()
 	if err != nil {
 		if errors.IsNotFound(err) {
-			_, err := resourceClient.Create(u, metav1.CreateOptions{})
+			logger.Info("Secret not found. Creating secret")
+			secretNew, err := resourceClient.Create(u, metav1.CreateOptions{})
 			if err != nil {
 				logger.Error(err, "Error creating secret")
 				return nil, err
 			}
 			logger.Info("Secret created")
-			return u, nil
+			return secretNew, nil
 		}
 		return nil, err
 	}
+
 	existingSecretData, _, _ := unstructured.NestedMap(existingSecret.Object, "data")
 	payloadInterim := make(map[string]interface{})
-	for k, v := range payload {
+	for k, v := range s.data {
 		payloadInterim[k] = base64.StdEncoding.EncodeToString(v)
 	}
 	comparisonResult := nestedMapComparison(existingSecretData, payloadInterim)
-	if comparisonResult.Success {
-		logger.Debug("Secret data is same. Skip Update")
-	} else {
-		logger.Info("Secret data is different; update secret", "Diff", comparisonResult.Diff)
-		_, err = resourceClient.Update(u, metav1.UpdateOptions{})
+	if !comparisonResult.Success {
+		logger.Info("Secret data is different", "Diff", comparisonResult.Diff)
+		secretNew, err := resourceClient.Create(u, metav1.CreateOptions{})
```

Comment:
Here it should call `resourceClient.Update` instead of `Create`, otherwise `AlreadyExists` error will be returned if trying to create an already existing resource.

Reply:
Done